app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt

```diff
@@ -45,41 +47,45 @@ fun HomeScreen(
         }
     }
     LaunchedEffect(viewModel) {
-        viewModel.navigator.collect { destination -> navigator(destination) }
+        viewModel.output.navigator.collect { destination -> navigator(destination) }
     }
 
-    val showMyCoinsLoading: IsLoading by viewModel.showMyCoinsLoading.collectAsState()
-    val showTrendingCoinsLoading: IsLoading by viewModel.showTrendingCoinsLoading.collectAsState()
-    val myCoins: List<CoinItemUiModel> by viewModel.myCoins.collectAsState()
-    val trendingCoins: List<CoinItemUiModel> by viewModel.trendingCoins.collectAsState()
+    val showMyCoinsLoading: IsLoading by viewModel.output.showMyCoinsLoading.collectAsState()
+    val showTrendingCoinsLoading: LoadingState by viewModel.output.showTrendingCoinsLoading.collectAsState()
+    val myCoins: List<CoinItemUiModel> by viewModel.output.myCoins.collectAsState()
+    val trendingCoins: List<CoinItemUiModel> by viewModel.output.trendingCoins.collectAsState()
 
     HomeScreenContent(
         showMyCoinsLoading = showMyCoinsLoading,
         showTrendingCoinsLoading = showTrendingCoinsLoading,
         myCoins = myCoins,
         trendingCoins = trendingCoins,
         onMyCoinsItemClick = viewModel.input::onMyCoinsItemClick,
-        onTrendingItemClick = viewModel.input::onTrendingCoinsItemClick
+        onTrendingItemClick = viewModel.input::onTrendingCoinsItemClick,
+        onTrendingCoinsLoadMore = { viewModel.input.getTrendingCoins(loadMore = true) }
     )
 }
 
 @Suppress("LongParameterList")
 @Composable
 private fun HomeScreenContent(
     showMyCoinsLoading: IsLoading,
-    showTrendingCoinsLoading: IsLoading,
+    showTrendingCoinsLoading: LoadingState,
     myCoins: List<CoinItemUiModel>,
     trendingCoins: List<CoinItemUiModel>,
-    onMyCoinsItemClick: (CoinItemUiModel) -> Unit,
-    onTrendingItemClick: (CoinItemUiModel) -> Unit
+    onMyCoinsItemClick: ((CoinItemUiModel) -> Unit)? = null,
+    onTrendingItemClick: ((CoinItemUiModel) -> Unit)? = null,
+    onTrendingCoinsLoadMore: (() -> Unit)? = null
 ) {
+    val trendingCoinsLastIndex = trendingCoins.lastIndex
+    val trendingCoinsState = rememberLazyListState()
```

Comment:
I guess we could apply the same approach to horizontal list, couldn't we?

Reply:
@hoangnguyen92dn Yes, we can apply it on `LazyColumn` as well. However we don't do it in this PR 💅